test/spec.controller.js

```diff
@@ -64,63 +163,210 @@ describe('Form Controller', function () {
+        it('skips setting errorValues if there are no form values', () => {
+            controller.setErrors({}, req, res);
+            req.sessionModel.toJSON().should.not.contain.keys('errorValues');
         });
 
-        it('passes through to parent errorHandler for all other errors', function () {
-            var err = new Error('foo');
-            controller.errorHandler(err, req, res, next);
-            Form.prototype.errorHandler.should.have.been.calledWithExactly(err, req, res, next);
-            Form.prototype.errorHandler.should.have.been.calledOn(controller);
+        it('sets error to session model', () => {
+            let err = {};
+            controller.setErrors(err, req, res);
+            req.sessionModel.get('errors').should.equal(err);
+        });
+    });
+
+    describe('locals', () => {
+        let controller;
+
+        beforeEach(() => {
+            controller = new Controller(options);
+            sinon.stub(Controller.prototype, 'getNextStep').returns('/next/step');
+        });
+
+        afterEach(() => {
+            Controller.prototype.getNextStep.restore();
         });
 
+        it('returns initial locals', () => {
+            req.baseUrl = '/base';
+            controller.locals(req, res).should.deep.equal({
+                baseUrl: '/base',
+                nextPage: '/next/step'
+            });
+
+        });
     });
 
-    describe('missingPrereqHandler', function () {
+    describe('missingPrereqHandler', () => {
+        let controller;
 
-        beforeEach(function () {
-            controller.options.steps = {
+        beforeEach(() => {
+            options.steps = {
                 '/one': { next: '/two' },
                 '/two': { next: '/three' },
                 '/three': { next: '/four' },
                 '/four': {}
             };
+            controller = new Controller(options);
         });
 
-        it('redirects to the step following the most recently completed', function () {
+        it('redirects to the step following the most recently completed', () => {
             req.sessionModel.set('steps', ['/one']);
             controller.missingPrereqHandler(req, res, next);
             res.redirect.should.have.been.calledWith('/two');
         });
 
-        it('redirects to the first step if no steps have been completed', function () {
+        it('redirects to the most recently completed if it has no next step', () => {
+            req.sessionModel.set('steps', ['/four']);
+            controller.missingPrereqHandler(req, res, next);
+            res.redirect.should.have.been.calledWith('/four');
+        });
+
+        it('redirects to the first step if no steps have been completed', () => {
             req.sessionModel.set('steps', []);
             controller.missingPrereqHandler(req, res, next);
             res.redirect.should.have.been.calledWith('/one');
         });
 
```

Comment:
We could do, but that feels like implementation and as long as this function is returning the correct value that should be fine